platform/database/sandbox.go

```diff
@@ -0,0 +1,198 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/GuiaBolso/darwin"
+	"github.com/fewlinesco/go-pkg/platform/metrics"
+	"github.com/jmoiron/sqlx"
+)
+
+type sandboxDB struct {
+	db *sqlx.DB
+	tx *sqlx.Tx
+}
+
+type sandboxTx struct {
+	tx            *sqlx.Tx
+	savepointName string
+}
+
+func TestConnect(config Config) (DB, error) {
+	db, err := connect(config)
+	if err != nil {
+		return nil, fmt.Errorf("can't connect to database: %v", err)
+	}
+
+	tx, err := db.Beginx()
+	if err != nil {
+		return nil, fmt.Errorf("could not create the test database transaction: %w", err)
+	}
+
+	return &sandboxDB{
+		db: db,
+		tx: tx,
+	}, nil
+}
+
+func (db *sandboxDB) Close() error {
+	err := db.tx.Rollback()
+	closeErr := db.db.Close()
+	if err != nil {
+		return fmt.Errorf("unable to rollback test transaction: %w", err)
+	}
+	if closeErr != nil {
+		return fmt.Errorf("unable to close database connection: %w", closeErr)
+	}
+	return nil
+}
+
+func (db *sandboxDB) NewGenericDriver(dialect darwin.Dialect) *darwin.GenericDriver {
+	panic("test database cannot produce a darwin driver")
+}
+
+func (db *sandboxDB) Begin() (Tx, error) {
+	fmt.Println("BEGIN")
+	db.tx.MustExec("SAVEPOINT postgres_savepoint;")
+
+	return &sandboxTx{tx: db.tx}, nil
+}
+
+func (db *sandboxDB) SelectContext(ctx context.Context, dest interface{}, statement string, args ...interface{}) error {
+	var err error
+
+	metrics.RecordElapsedTimeInMilliseconds(ctx, metricQueryLatencyMs, func() {
```

Comment:
Q: Should we keep the metrics recorders for during the tests?

Reply:
this is very true. 
I still have to reform a bit this part since it's lacking stuff and this is the result of heavy copy paste and adapt. But thanks for the notice :)